File: 55. Jump Game/Solution.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        zeros = []
        for i in range(len(nums)-1):
            if (nums[i] == 0):
                zeros.append(i)
        for i in range(len(zeros)):
            for j in range(zeros[i]-1, -1, -1):
                if (nums[j]+j > zeros[i]):
                    zeros[i] = -1
                    break
        for zero in zeros:
            if zero != -1:
                return False
        return True

File: 55. Jump Game/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {

    public boolean canJump(int[] nums) {
        List<Integer> zeros = new ArrayList<>();
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] == 0)
                zeros.add(i);
        }
        for (int i = 0; i < zeros.size(); i++) {
            for (int j = zeros.get(i) - 1; j >= 0; j--) {
                if (j + nums[j] > zeros.get(i)) {
                    zeros.remove(i);
                    i--;
                    break;
                }
            }
            if (zeros.isEmpty())
                break;
        }
        return zeros.isEmpty();
    }
}